test/fulfillment_order_test.rb

```diff
@@ -132,5 +132,173 @@ def setup
         assert_equal 'closed', fulfillment_order.status
       end
     end
+
+    context "#fulfillment_request" do
+      should "make a fulfillment request for a fulfillment order" do
+        original_fulfillment_order = ActiveSupport::JSON.decode(load_fixture('fulfillment_order'))
+        submitted_fulfillment_order = original_fulfillment_order.clone
+        submitted_fulfillment_order['id'] = 2
+        submitted_fulfillment_order['status'] = 'open'
+        submitted_fulfillment_order['request_status'] = 'submitted'
+        unsubmitted_fulfillment_order = original_fulfillment_order.clone
+        unsubmitted_fulfillment_order['id'] = 3
+        unsubmitted_fulfillment_order['request_status'] = 'unsubmitted'
+        original_fulfillment_order['status'] = 'in_progress'
+        body = {
+          original_fulfillment_order: original_fulfillment_order,
+          submitted_fulfillment_order: submitted_fulfillment_order,
+          unsubmitted_fulfillment_order: unsubmitted_fulfillment_order
+        }
+        request_body = {
+          fulfillment_request: {
+            fulfillment_order_line_items: [
+              { id: 1, quantity: 1 }
+            ],
+            message: 'Fulfill this FO, please.'
+          }
+        }
+        fake "fulfillment_orders/519788021/fulfillment_request", :method => :post,
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode(body)
+
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+        params = {
+          fulfillment_order_line_items: [{ id: 1, quantity: 1 }],
+          message: "Fulfill this FO, please."
+        }
+        original_submitted_unsubmitted_fos = fulfillment_order.fulfillment_request(params)
+
+        assert_equal 'in_progress', fulfillment_order.status
+
+        original_fo = original_submitted_unsubmitted_fos['original_fulfillment_order']
+        assert_equal 519788021, original_fo.id
+        assert_equal 'in_progress', original_fo.status
+
+        submitted_fo = original_submitted_unsubmitted_fos['submitted_fulfillment_order']
+        assert_equal 2, submitted_fo.id
+        assert_equal 'open', submitted_fo.status
+        assert_equal 'submitted', submitted_fo.request_status
+
+        unsubmitted_fo = original_submitted_unsubmitted_fos['unsubmitted_fulfillment_order']
+        assert_equal 3, unsubmitted_fo.id
+        assert_equal 'open', unsubmitted_fo.status
+        assert_equal 'unsubmitted', unsubmitted_fo.request_status
+      end
+    end
```

Comment:
It's also possible for the response to be something like:
```
{
  original_fulfillment_order: {...},
  submitted_fulfillment_order: {...exact same as original...},
  unsubmitted_fulfillment_order: nil,
}
```
This case should also be tested.

Reply:
Added above.